SEQLinkage/Core.py

```diff
@@ -692,7 +757,61 @@ def run_each_region(regions,data,extractor,maker,writer):
             with env.success_counter.get_lock():
                 env.success_counter.value += 1
             results[region[3]]=maker.dtest[region[3]]
-    env.log('write to pickle')
-    import pickle
-    with open(os.path.join(env.tmp_cache,env.output+'.pickle'), 'wb') as handle:
-        pickle.dump(results, handle, protocol=pickle.HIGHEST_PROTOCOL)
\ No newline at end of file
+            if len(results)==25:
```

Comment:
we are using this mechanism to cache and prevent too much data in the memory? Let's make it a function parameter "cache_size = 25" with default 25.

Reply:
will do late.

Reply:
updated